core/trino-main/src/main/java/io/trino/SystemSessionProperties.java

```diff
@@ -1173,4 +1182,18 @@ public static boolean isLegacyCatalogRoles(Session session)
     {
         return session.getSystemProperty(LEGACY_CATALOG_ROLES, Boolean.class);
     }
+
+    public static RetryPolicy getRetryPolicy(Session session)
+    {
+        RetryPolicy retryPolicy = session.getSystemProperty(RETRY_POLICY, RetryPolicy.class);
+        if (retryPolicy != RetryPolicy.NONE) {
```

Comment:
It feels getter is not the best place to validation, but I am not sure if we have better option right now.
cc: @findepi

Reply:
We do validation in other getters, it feels like it should be fine.